packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -230,36 +230,78 @@ class _RawMenuAnchorScope extends InheritedWidget {
 ///
 /// ** See code in examples/api/lib/widgets/raw_menu_anchor/raw_menu_anchor.1.dart **
 /// {@end-tool}
-///
-class RawMenuAnchor extends StatelessWidget {
```

Comment:
Previously, `RawMenuAnchor` would either build a `_RawMenuAnchorNode` or a `_RawMenuAnchorOverlay` based on which constructor was used; I factored out the build by making them redirecting constructors.

```diff
- class RawMenuAnchor extends StatelessWidget {
- 
-   const RawMenuAnchor(...);
-   const RawMenuAnchor.overlayBuilder(...);
-   const RawMenuAnchor.node(...);
- 
- }

+ sealed class RawMenuAnchor extends StatefulWidget {
- sealed class _RawMenuAnchor extends StatefulWidget {
+   const factory RawMenuAnchor(...);
+   const factory RawMenuAnchor.overlayBuilder(...);
+   const factory RawMenuAnchor.node(...);
  }
```

Then I made `RawMenuAchor` concrete by combining with the `node` class:

```dart
class RawMenuAnchor extends StatefulWidget {
   const factory RawMenuAnchor(…);
   const factory RawMenuAnchor.overlayBuilder(…);
   const RawMenuAnchor.node(…);

  // subclasses can override this
  State<RawMenuAnchor> createState() => _RawMenuAnchorNodeState();
}
```